### DISCUSSION TITLE
Plans for source code control in PowerApps?

### DISCUSSION THREAD

**Discussion by arms3** [microsoft/PowerApps-Language-Tooling#711] [Q&A]
Hey I'm a little confused with the plans for source control in PowerApps. I see the pac cli pack / unpack commands are being deprecated, do these currently rely on the PASopa library or the Persistence library?

We have production workloads that rely on being able to manage source code for merging and deployment, and I don't see a clear answer to whether we should be planning to retool our pipelines based on the new source control feature demoed in May but still apparently missing.

Should I invest time in investigating the Persistence library? Can we expect the source control feature soon?

**Comment by BetimBeja** [comment]
@arms3  Check this out https://learn.microsoft.com/en-us/power-platform/release-plan/2024wave2/power-apps/manage-source-code-canvas-apps

**Reply by arms3** [reply]
OK thanks I'll try the wave 2 update now then. The wording on that page is super confusing and contradictory though!

> You can use Power Platform CLI to convert a .msapp file into source code files. With this feature, you no longer need to use the Power Platform CLI to convert the .msapp files.

**Reply by arms3** [reply]
Hmm it's still not there in my environment when wave 2 is enabled so it's unclear how the feature will work and when it's planned for release.

**Comment by marcelbf** [comment]
To source control your canvas app, please use the [Power Platform git integration](https://learn.microsoft.com/power-platform/alm/git-integration/canvas-apps-git-integration).

More information about the [YAML serialization format](https://learn.microsoft.com/power-apps/maker/canvas-apps/power-apps-yaml#power-apps-yaml-schema-versions) is available in the documentation.
